docker/hq-compose-os-macos-m1-12.yml

```diff
@@ -0,0 +1,18 @@
+# Service configurations specific to 'arm64' (Apple M1) architecture for MacOS Monterey (v 12).
+
+version: '2.3'
+
+services:
+  postgres:
+    image: dimagi/docker-postgresql
+    platform: linux/amd64
```

Comment:
Hey @biyeun 

I am trying to refactor this to avoid multiple files by using compose environment variables (that depend on the OS type).

If the OS is linux or M1-11, the platform is not set in the hq-services.yml file, what should be the platform values for M1-11 and the default OS

Reply:
If you have a platform that doesn't have a configuration in these compose files, I think that means nobody has needed it yet. If you have such a platform, you will need to figure out what works and a submit a PR for it.

Reply:
> I am trying to refactor this to avoid multiple files

Just an FYI, the multiple files was very much intentional since even seemingly similar systems (e.g. `M1-11` vs `M1-12`) required different configurations. The value of multiple files is that it makes it easier to maintain. That is, easier to both:
1. Understand the differences at a quick glance, and
2. remove old configurations when a platform is no longer needed.